Rustc suggests both std and core imports at times

rustc often offers a help to import something from std and core:

```rust
error[E0433]: failed to resolve: use of undeclared type `NonZeroU32`
  --> ...\test.rs:20:33
   |
20 | ...                   NonZeroU32::new(5).unwrap(),
   |                       ^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
6  | use core::num::NonZeroU32;
   |
6  | use std::num::NonZeroU32;
   |
```

Perhaps in such situations rustc could use an heuristic to suggest only one of the two imports.

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -1707,6 +1707,12 @@ crate fn show_candidates(
         candidates.iter().map(|c| path_names_to_string(&c.path)).collect();
 
     path_strings.sort();
+    let core_path_strings =
+        path_strings.iter().filter(|p| p.starts_with("core::")).cloned().collect::<Vec<String>>();
+    if !core_path_strings.is_empty() {
+        path_strings.retain(|p| !p.starts_with("core::"));
+    }
+    path_strings.extend(core_path_strings);
     path_strings.dedup();
 
     let (determiner, kind) = if candidates.len() == 1 {

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -1708,10 +1708,7 @@ crate fn show_candidates(
 
     path_strings.sort();
     let core_path_strings =
-        path_strings.iter().filter(|p| p.starts_with("core::")).cloned().collect::<Vec<String>>();
-    if !core_path_strings.is_empty() {
-        path_strings.retain(|p| !p.starts_with("core::"));
-    }
+        path_strings.drain_filter(|p| p.starts_with("core::")).collect::<Vec<String>>();
     path_strings.extend(core_path_strings);
     path_strings.dedup();
 

--- a/compiler/rustc_resolve/src/lib.rs
+++ b/compiler/rustc_resolve/src/lib.rs
@@ -10,6 +10,7 @@
 
 #![doc(html_root_url = "https://doc.rust-lang.org/nightly/nightly-rustc/")]
 #![feature(box_patterns)]
+#![feature(drain_filter)]
 #![feature(bool_to_option)]
 #![feature(crate_visibility_modifier)]
 #![feature(format_args_capture)]

